pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -59,9 +65,14 @@
   protected HelixManager _helixManager;
   protected String _authToken;
 
+  // Fixed size LRU cache with InstanceName - SegmentName pair as key, and segment related
+  // errors as the value.
+  protected LoadingCache<Pair<String, String>, SegmentErrorInfo> _errorCache;
```

Comment:
Why do we need instance name in every key here? BaseTableDataManager lives on an instance, so this part of the key  will be sam for every segment.

Reply:
Good question. This is to avoid single table hosting multiple tables. The size of this cache is per instance, as opposed to per table (which could blow up in MT case).

Reply:
I think you mean single instance hosting multiple tables, in which case, the comment in line 68 should be "tableName - segmetName" pair as opposed to "instanceName - segmentName" pair?